File: PolicyPlus/LoadedAdmx.vb
Public Class LoadedAdmx
    Dim Bundle As AdmxBundle
    Public Sub PresentDialog(Workspace As AdmxBundle)
        LsvAdmx.Items.Clear()
        For Each admx In Workspace.Sources.Keys
            Dim lsvi = LsvAdmx.Items.Add(IO.Path.GetFileName(admx.SourceFile))
            lsvi.SubItems.Add(IO.Path.GetDirectoryName(admx.SourceFile))
            lsvi.SubItems.Add(admx.AdmxNamespace)
            lsvi.Tag = admx
        Next
        LoadedAdmx_SizeChanged(Nothing, Nothing)
        ChNamespace.Width -= SystemInformation.VerticalScrollBarWidth ' For some reason, this only needs to be taken into account on the first draw
        Bundle = Workspace
        ShowDialog()
    End Sub
    Private Sub LsvAdmx_DoubleClick(sender As Object, e As EventArgs) Handles LsvAdmx.DoubleClick
        DetailAdmx.PresentDialog(LsvAdmx.SelectedItems(0).Tag, Bundle)
    End Sub
    Private Sub LoadedAdmx_SizeChanged(sender As Object, e As EventArgs) Handles Me.SizeChanged
        ChNamespace.Width = Math.Max(30, LsvAdmx.ClientRectangle.Width - ChFolder.Width - ChFileTitle.Width)
    End Sub
    Private Sub LsvAdmx_KeyDown(sender As Object, e As KeyEventArgs) Handles LsvAdmx.KeyDown
        If e.KeyCode = Keys.Enter And LsvAdmx.SelectedItems.Count > 0 Then LsvAdmx_DoubleClick(sender, e)
    End Sub
End Class

File: PolicyPlus/LoadedAdmx.cs
using System;
using System.Windows.Forms;

namespace PolicyPlus
{
    public partial class LoadedAdmx
    {
        private AdmxBundle Bundle;

        public LoadedAdmx()
        {
            InitializeComponent();
        }
        public void PresentDialog(AdmxBundle Workspace)
        {
            LsvAdmx.Items.Clear();
            foreach (var admx in Workspace.Sources.Keys)
            {
                var lsvi = LsvAdmx.Items.Add(System.IO.Path.GetFileName(admx.SourceFile));
                lsvi.SubItems.Add(System.IO.Path.GetDirectoryName(admx.SourceFile));
                lsvi.SubItems.Add(admx.AdmxNamespace);
                lsvi.Tag = admx;
            }
            LoadedAdmx_SizeChanged(null, null);
            ChNamespace.Width -= SystemInformation.VerticalScrollBarWidth; // For some reason, this only needs to be taken into account on the first draw
            Bundle = Workspace;
            ShowDialog();
        }
        private void LsvAdmx_DoubleClick(object sender, EventArgs e)
        {
            My.MyProject.Forms.DetailAdmx.PresentDialog((AdmxFile)LsvAdmx.SelectedItems[0].Tag, Bundle);
        }
        private void LoadedAdmx_SizeChanged(object sender, EventArgs e)
        {
            ChNamespace.Width = Math.Max(30, LsvAdmx.ClientRectangle.Width - ChFolder.Width - ChFileTitle.Width);
        }
        private void LsvAdmx_KeyDown(object sender, KeyEventArgs e)
        {
            if (e.KeyCode == Keys.Enter & LsvAdmx.SelectedItems.Count > 0)
                LsvAdmx_DoubleClick(sender, e);
        }
    }
}